docs/extension-development/check-if-woo-is-active.md

```diff
@@ -4,28 +4,45 @@ menu_title: Check if WooCommerce is active
 tags: how-to
 ---
 
-You can wrap your plugin in a check to see if WooCommerce is installed:
+When developing for WooCommerce, it's important to check that WooCommerce is installed and active before your own code runs. This ensures no errors occur from missing WooCommerce functions or classes.
+
+There are a few methods to achieve this. The most reliable approach is to initiate your code on `plugins_loaded`. This means you can be sure that WooCommerce has already initiated, by checking if one of its core classes exists.
 
 ```php
-// Test to see if WooCommerce is active (including network activated).
+add_action( 'plugins_loaded', 'prefix_plugins_loaded' );
 
-$plugin_path = trailingslashit( WP_PLUGIN_DIR ) . 'woocommerce/woocommerce.php';
+function prefix_plugins_loaded() {
+	// Check if the "WooCommerce" class exists. If not, exit early.
+	if ( ! class_exists( 'WooCommerce' ) ) {
+		return;
+	}
 
-if (
+	// Custom code here. WooCommerce is active and initiated...
+}
+```
 
-in_array( $plugin_path, wp_get_active_and_valid_plugins() )
+Another route you could take is to initiate your code on the `woocommerce_init` hook. We know this only runs when WooCommerce is active and initiated.
 
-|| in_array( $plugin_path, wp_get_active_network_plugins() )
+```php
+add_action( 'woocommerce_init', 'prefix_woocommerce_init' );
 
-) {
+function prefix_woocommerce_init() {
+	// Custom code here. WooCommerce is active and initiated...
+}
+```
 
-// Custom code here. WooCommerce is active, however it has not
+However, in some cases you may want to check if WooCommerce exists *before* `plugins_loaded`. This method is slightly less reliable, because it assumes the path of the WooCommerce plugin hasn't been changed. It's also worth noting that this method may run _before or after_ WooCommerce is ready; meaning you may not be able to utilize WooCommerce functions yet.
 
-// necessarily initialized (when that is important, consider
+```php
+// Test to see if WooCommerce is active (including network activated).
 
-// using the \`woocommerce_init\` action).
+$plugin_path = trailingslashit( WP_PLUGIN_DIR ) . 'woocommerce/woocommerce.php';
 
+if (
+	in_array( $plugin_path, wp_get_active_and_valid_plugins() ) ||
+	in_array( $plugin_path, wp_get_active_network_plugins() )
+) {
+	// Custom code here. WooCommerce is active, however it has not
+	// necessarily initialized...
```

Comment:
I'd prefer to leave this out of the docs. I don't know of any case where this needs to happen sooner than `plugins_loaded` and the caveats are too risky. If you know of a really great reason for leaving it in, I'm all ears :)

Reply:
1wp_get_active_and_valid_plugins()` and `wp_get_active_network_plugins()` are intended to be private functions. I don't think these should be encouraged